core/domain/collection_domain.py

```diff
@@ -1303,3 +1303,15 @@ def does_user_have_any_role(self, user_id):
             user_id in self.editor_ids or
             user_id in self.viewer_ids
         )
+
+    def add_new_contribution_for_user(self, contributor_id):
+        """Add a new contributor to the contributors summary.
+
+        Args:
+            contributor_id: str. ID of the contributor to be added.
+        """
+        if contributor_id not in constants.SYSTEM_USER_IDS:
```

Comment:
What happens if contributor_id is part of `constants.SYSTEM_USER_IDS`, is it handled elsewhere?

Reply:
We don't want to record the contributions for system users since these are just automatic scripts and bots.

Reply:
Added a comment to make this explicit.